src/test/java/org/blyznytsia/scanner/ComponentAnnotationScannerTest.java

```diff
@@ -0,0 +1,44 @@
+package org.blyznytsia.scanner;
+
+
+import org.assertj.core.api.Assertions;
+import org.blyznytsia.model.BeanDefinition;
+import org.blyznytsia.scanner.data.TestService1;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+class ComponentAnnotationScannerTest {
+
+    private static final String PACKAGE_NAME = "org.blyznytsia.scanner.data";
+    private static final String SERVICE1_NAME = "anotherNameService";
+    private static final String SERVICE2_NAME = "testService2";
+    private static final int BEAN_DEFINITIONS_SIZE = 2;
+    private static final int SERVICE2_DEPENDENCIES_SIZE = 1;
+    private static final String SERVICE2_DEPENDENCY_NAME = TestService1.class.getName();
+
+    @Test
+    void scan_givenValidPackage_shouldReturnValidBeanDefinitionsList() {
+        final var beanDefinitions =
+                new ComponentAnnotationScanner().scan(PACKAGE_NAME);
+        beanDefinitions.forEach(System.out::println);
+        Assertions.assertThat(checkFullPresence(beanDefinitions, SERVICE2_NAME, SERVICE1_NAME)).isTrue();
+
+        final var service1 = getBeanDefinitionByName(beanDefinitions, SERVICE1_NAME);
```

Comment:
I suppose we no need to overuse the `final` word, except in the case where it is vital